pkg/collector/corechecks/containers/docker.go

```diff
@@ -184,6 +178,16 @@ func (d *DockerCheck) Run() error {
 		if err != nil {
 			log.Errorf("Could not collect tags for container %s: %s", c.ID[:12], err)
 		}
+		// Track image_name and image_tag tags by image for use in countAndWeightImages
+		for _, t := range tags {
+			if strings.HasPrefix(t, "image_name:") || strings.HasPrefix(t, "image_tag:") {
+				if _, found := imageTagsByImageID[c.ImageID]; !found {
+					imageTagsByImageID[c.ImageID] = []string{t}
+				} else {
+					imageTagsByImageID[c.ImageID] = append(imageTagsByImageID[c.ImageID], t)
+				}
+			}
+		}
```

Comment:
This was added because I noticed we are associating `RepoTag[0]` in these metrics as well, and thought we should instead be consistent in the tags used. Open to hearing other thoughts though if this is not necessary/out of scope.

Reply:
So we're using tags value to retrieve what was set before by `ResolveImageNameFromContainer()`. Though we have access to the Container object, no? (`c for cList`).
It's probably cleaner to call `ResolveImageNameFromContainer()`. Looking through tags might be faster but normally result form `ResolveImageNameFromContainer()` is cached.

Reply:
unfortunately `cList` is a different container object (`containers.Container` vs. `types.ContainerJSON`), and the former doesn't have `config.Image` in the struct